packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts

```diff
@@ -108,6 +123,52 @@ abstract class VirtualNodeBase extends cdk.Resource implements IVirtualNode {
    * The Mesh which the VirtualNode belongs to
    */
   public abstract readonly mesh: IMesh;
+
+  /**
+   * Grants the given entity all read permissions for this VirtualNode.
+   */
+  public grantRead(identity: iam.IGrantable): iam.Grant {
+    return this.grant(identity,
+      'appmesh:DescribeVirtualNode',
+      'appmesh:ListVirtualNode',
+      'appmesh:ListTagsForResource',
+    );
+  }
+
+  /**
+   * Grants the given entity all write permissions for this VirtualNode.
+   */
+  public grantWrite(identity: iam.IGrantable): iam.Grant {
+    return this.grant(identity,
+      'appmesh:CreateVirtualNode',
+      'appmesh:UpdateVirtualNode',
+      'appmesh:DeleteVirtualNode',
+      'appmesh:TagResource',
+      'appmesh:UntagResource',
+    );
+  }
+
+  /**
+   * Grants the given entity `appmesh:StreamAggregatedResources`.
+   */
+  public grantStreamAggregatedResources(identity: iam.IGrantable): iam.Grant {
+    return iam.Grant.addToPrincipal({
+      grantee: identity,
+      actions: ['appmesh:StreamAggregatedResources'],
+      resourceArns: [this.virtualNodeArn],
+    });
```

Comment:
Shouldn't this be

```suggestion
    return this.grant(identity, 'appmesh:StreamAggregatedResources');
```

?